src/LocaleService.php

```diff
@@ -194,4 +194,128 @@ public function setSystemLocale(string $locale = null)
             setlocale(LC_ALL, ...$systemConfig);
         }
     }
+
+    /**
+     * Return the current locale name
+     * @return string
+     * @throws \Exolnet\Translation\TranslationException
+     */
+    public function getCurrentLocaleName()
+    {
+        return $this->getLocaleName($this->getCurrentLocale());
+    }
+
+    /**
+     * Return the locale name for a specified locale
+     * @param string|null $locale
+     * @return string
+     * @throws \Exolnet\Translation\TranslationException
+     */
+    public function getLocaleName(string $locale = null)
+    {
+        if (!$locale) {
+            $locale = $this->getCurrentLocale();
+        }
+
+        if (!isset($this->getLocalesAvailable()[$locale])) {
+            return '';
+        }
+
+        return array_get($this->getLocalesAvailable()[$locale], 'name', '');
+    }
+
+    /**
+     * Return the current locale native name
+     * @return string
+     * @throws \Exolnet\Translation\TranslationException
+     */
+    public function getCurrentLocaleNativeName()
+    {
+        return $this->getLocaleNativeName($this->getCurrentLocale());
+    }
+
+    /**
+     * Return the locale native name for a specified locale
+     * @param string|null $locale
+     * @return string
+     * @throws \Exolnet\Translation\TranslationException
+     */
+    public function getLocaleNativeName(string $locale = null)
+    {
+        if (!$locale) {
+            $locale = $this->getCurrentLocale();
+        }
+
+        if (!isset($this->getLocalesAvailable()[$locale])) {
+            return '';
```

Comment:
Is it a better idea to fallback on the `name` using `getLocaleName` method?

Reply:
Should be fixed